File: Fun stuff non-related to beaglebone/Pig Game JavaScript/piggame.js
// Pig Game for two players, playable in a web browser

let scores = [0, 0];
let currentPlayer = 0;
let roundTotal = 0;
let gameActive = true;

function updateUI() {
    document.getElementById("score1").textContent = scores[0];
    document.getElementById("score2").textContent = scores[1];
    document.getElementById("current-player").textContent = `Spelare ${currentPlayer + 1}`;
    document.getElementById("round-total").textContent = roundTotal;
}

function rollDice() {
    if (!gameActive) return;
    let num = Math.floor(Math.random() * 6) + 1;
    document.getElementById("dice").textContent = `Du kastade: ${num}`;
    if (num === 1) {
        scores[currentPlayer] = 0;
        roundTotal = 0;
        alert(`Spelare ${currentPlayer + 1}: Du tappade ALLA dina poäng!`);
        nextPlayer();
    } else {
        roundTotal += num;
        updateUI();
    }
}

function savePoints() {
    if (!gameActive) return;
    scores[currentPlayer] += roundTotal;
    roundTotal = 0;
    updateUI();
    if (scores[currentPlayer] >= 100) {
        gameActive = false;
        alert(`Grattis Spelare ${currentPlayer + 1}, du vann med ${scores[currentPlayer]} poäng!`);
    } else {
        nextPlayer();
    }
}

function nextPlayer() {
    roundTotal = 0;
    currentPlayer = currentPlayer === 0 ? 1 : 0;
    updateUI();
}

function resetGame() {
    scores = [0, 0];
    currentPlayer = 0;
    roundTotal = 0;
    gameActive = true;
    document.getElementById("dice").textContent = "";
    updateUI();
}

window.onload = function() {
    document.body.innerHTML = `
        <h2>Pig Game - Två spelare</h2>
        <div>
            <strong>Spelare 1 Poäng:</strong> <span id="score1">0</span><br>
            <strong>Spelare 2 Poäng:</strong> <span id="score2">0</span><br>
            <strong>Nuvarande spelare:</strong> <span id="current-player">Spelare 1</span><br>
            <strong>Summa denna runda:</strong> <span id="round-total">0</span><br>
            <span id="dice"></span>
        </div>
        <button onclick="rollDice()">Kasta tärningen</button>
        <button onclick="savePoints()">Spara poäng</button>
        <button onclick="resetGame()">Starta om</button>
    `;
    updateUI();
};

File: python/games/pig_game.py
import random

def kasta(player):
    total = 0
    while True:
        num = random.randint(1, 6)
        total += num
        if num == 1:
            print(f"{player}: Sorry, du tappade dina poäng för denna runda!")
            return 0
        print(f"{player}: Summa för detta varv: {total}")
        val = input(f"{player}: Tryck på k för att kasta på nytt eller s för att spara: ")
        match val:
            case "k":
                continue
            case "s":
                return total

def main():
    players = ["Spelare 1", "Spelare 2"]
    scores = {p: 0 for p in players}
    turn = 0

    while True:
        current_player = players[turn % 2]
        print(f"\n{current_player}s tur. Nuvarande poäng: {scores[current_player]}")
        round_score = kasta(current_player)
        scores[current_player] += round_score
        print(f"{current_player} har nu {scores[current_player]} poäng")
        if scores[current_player] >= 100:
            print(f"Grattis {current_player}, du vann med {scores[current_player]} poäng!")
            break
        input("Tryck 'Enter' för att fortsätta till nästa spelare")
        turn += 1

main()